spec/serialization_spec.rb

```diff
@@ -43,6 +45,505 @@ def self.name
       expect(json['data'][0]['attributes']).to eq('first_name' => 'John')
     end
 
+    describe 'types' do
+      # can coerce everything...
+      context 'when string' do
+        before do
+          resource.attribute :age, :string
+        end
+
+        it 'coerces' do
+          PORO::Employee.create(age: 1)
+          render
+          expect(attributes['age']).to eq('1')
+        end
+      end
+
+      context 'when integer' do
+        before do
+          resource.attribute :age, :integer
+        end
+
+        it 'coerces strings' do
+          PORO::Employee.create(age: '40')
+          render
+          expect(attributes['age']).to eq(40)
+        end
+
+        it 'allows nils' do
+          PORO::Employee.create(age: nil)
+          render
+          expect(attributes['age']).to eq(nil)
+        end
+
+        context 'when cannot coerce' do
+          before do
+            PORO::Employee.create(age: 'foo')
+          end
+
+          # Note ArgumentError not ConstraintError
+          # Unfortunate but not really our purview
+          it 'raises error' do
+            expect {
+              render
+            }.to raise_error(ArgumentError)
```

Comment:
Fine punting on this for a little while, but I think we will want to eventually wrap this in a more helpful suite-specific error.

Reply:
Did you see the commit message on why I am specifically not doing this?

Reply:
I did, but seems I glossed over: 

>  it would come at the cost of hiding real errors for custom types

which is a good enough reason to punt on this for the time being.